src/main/java/org/example/service/TableService.java

```diff
@@ -0,0 +1,43 @@
+package org.example.service;
+
+import org.example.model.Developer;
+
+import java.io.PrintWriter;
+import java.util.List;
+
+public class TableService {
+
+    private final DeveloperService developerService = DeveloperService.getInstance();
+
+    public void generateHtmlTable(PrintWriter printWriter) {
+        List<Developer> developers = developerService.getDevelopers();
+        printWriter.println("<html>");
+        printWriter.println("<head>");
+        printWriter.println("</head>");
+        printWriter.write("<center>");
+        printWriter.println("<body>");
+        printWriter.write("<table>");
+        printWriter.write("<th>user_id</th>");
+        printWriter.write("<th>firstName</th>");
+        printWriter.write("<th>lastName</th>");
+        printWriter.write("<th>age</th>");
+        printWriter.write("<th>progLang</th>");
+        printWriter.write("<tr>");
+
+        for (Developer developer : developers) {
+            printWriter.write("<tr>");
+            printWriter.write("<td>" + developer.getUserId() + "</td>");
+            printWriter.write("<td>" + developer.getFirstName() + "</td>");
+            printWriter.write("<td>" + developer.getSecondName() + "</td>");
+            printWriter.write("<td>" + developer.getAge() + "</td>");
+            printWriter.write("<td>" + developer.getProgLang() + "</td>");
+            printWriter.write("</tr>");
+        }
+
+        printWriter.write("</tr>");
+        printWriter.println("</table>");
+        printWriter.println("</body>");
+        printWriter.write("</center>");
+        printWriter.println("</html>");
```

Comment:
Firstly, create the result string and call the printWriter.write only once at the end.

Reply:
fix